Content.Server/Weapon/Ranged/Systems/GunSystem.cs

```diff
@@ -103,6 +103,12 @@ public override void Shoot(GunComponent gun, List<IShootable> ammo, EntityCoordi
                             shotProjectiles.Add(uid);
                         }
 
+                        //signal ChemicalAmmo to transfer solution
+                        RaiseLocalEvent(cartridge.Owner, new AmmoShotEvent()
+                        {
+                            FiredProjectiles = shotProjectiles,
+                        }, false);
+
```

Comment:
Kinda suss raising the same event on the gun and the cartridges but I guess we'll see if it causes problems.